core/src/test/java/io/jenkins/pluginhealth/scoring/probes/GitHubOpenIssuesProbeTest.java

```diff
@@ -0,0 +1,136 @@
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+import io.jenkins.pluginhealth.scoring.model.updatecenter.UpdateCenter;
+
+import org.junit.jupiter.api.Test;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+
+class GitHubOpenIssuesProbeTest extends AbstractProbeTest<GitHubOpenIssuesProbe> {
+    @Override
+    GitHubOpenIssuesProbe getSpy() {
+        return spy(GitHubOpenIssuesProbe.class);
+    }
+
+    @Test
+    void shouldNotRunWithInvalidProbeResultRequirement() {
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+
+        when(plugin.getDetails()).thenReturn(
+            Map.of(),
+            Map.of(
+                IssueTrackerDetectionProbe.KEY, ProbeResult.success(IssueTrackerDetectionProbe.KEY, "")
+            ),
+            Map.of(
+                SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, "")
+            ),
+            Map.of(
+                SCMLinkValidationProbe.KEY, ProbeResult.failure(SCMLinkValidationProbe.KEY, ""),
+                IssueTrackerDetectionProbe.KEY, ProbeResult.failure(IssueTrackerDetectionProbe.KEY, "")
+            ),
+            Map.of(
+                SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, ""),
+                IssueTrackerDetectionProbe.KEY, ProbeResult.failure(IssueTrackerDetectionProbe.KEY, "")
+            ),
+            Map.of(
+                SCMLinkValidationProbe.KEY, ProbeResult.failure(SCMLinkValidationProbe.KEY, ""),
+                IssueTrackerDetectionProbe.KEY, ProbeResult.success(IssueTrackerDetectionProbe.KEY, "")
+            )
+        );
+
+        final GitHubOpenIssuesProbe probe = getSpy();
+        assertThat(probe.apply(plugin, ctx))
+            .usingRecursiveComparison()
+            .comparingOnlyFields("id", "status", "message")
+            .isEqualTo(ProbeResult.error(GitHubOpenIssuesProbe.KEY, "github-open-issues does not meet the criteria to be executed on null"));
+        verify(probe, never()).doApply(plugin, ctx);
+    }
+
+    @Test
+    void shouldBeAbleToFindNumberOfOpenIssuesInGH() throws IOException {
+        final String pluginName = "cloudevents";
+        final String repository = "jenkinsci/" + pluginName + "-plugin";
+        final String scmLink = "https://github.com/" + repository;
+
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final GitHub gh = mock(GitHub.class);
+        final GHRepository ghRepository = mock(GHRepository.class);
+        final io.jenkins.pluginhealth.scoring.model.updatecenter.Plugin.IssueTrackers issueTrackerGithub = new io.jenkins.pluginhealth.scoring.model.updatecenter.Plugin.IssueTrackers("github", "https://github.com/" + repository + "/issues", "https://github.com/" + repository + "/issues/new/choose");
```

Comment:
Why not importing the type and using a fully qualified name here?

Reply:
For some reason, IntelliJ did not add a qualified import. Maybe because of the common `Plugin` class.

In the case of the common `Plugin` class name in UpdateCenter and Model, it did that. That is why I did not think much of it.